HACKING.md

```diff
@@ -0,0 +1,78 @@
+# Tips and tricks for hacking on Buck2
+
+## Building the code
+
+Buck2 is written in Rust, **and currently requires a nightly toolchain**. You
+need **[rustup](https://rustup.rs)** installed to provision the necessary
+version of `rustc` and `cargo` as a result.
+
+You can either build `buck2` from a clone of the Buck2 repo (this will cause
+rustup/cargo to install the right nightly version of `rustc`):
+
+```sh
+git clone https://github.com/facebook/buck2.git
+cd buck2/
+cargo install --path=app/buck2
+```
+
+Or alternatively install it directly from GitHub:
+
+```sh
+rustup install nightly-2023-03-07
+cargo +nightly-2023-03-07 install --git https://github.com/facebook/buck2.git buck2
+```
+
+Build uses prebuilt `protoc` binary from
+[protoc-bin-vendored](https://crates.io/crates/protoc-bin-vendored) crate. If
+these binaries to do not work on your machine, the path to `protoc` binary and
+protobuf include path can be specified via `BUCK2_BUILD_PROTOC` and
+`BUCK2_BUILD_PROTOC_INCLUDE` environment variables.
+
+### Using [Nix] to compile the source
+
+Most [Nix] users provision tools directly with Nix itself, rather than rustup;
```

Comment:
I think we probably want one or two words here to make it clear that Nix is not the recommended way to do this, but a possible alternative. We don't want people trying to install Buck2 to accidentally think they need Nix before they continue.